ray-operator/controllers/ray/expectations/scale_expectations_test.go

```diff
@@ -0,0 +1,167 @@
+	assert.NoError(t, fakeClient.Create(ctx, &testPods[0]), "Fail to create worker pod1")
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupA), true)
+
+	// Expect delete.
+	exp.ExpectScalePod(namespace, rayClusterName, groupA, testPods[0].Name, Delete)
+	exp.ExpectScalePod(namespace, rayClusterName, groupB, testPods[1].Name, Delete)
+	exp.ExpectScalePod(namespace, rayClusterName, groupB, testPods[2].Name, Delete)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupA), false)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupB), false)
+	assert.NoError(t, fakeClient.Delete(ctx, &testPods[1]), "Fail to delete worker pod2")
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupB), false)
+	assert.NoError(t, fakeClient.Delete(ctx, &testPods[2]), "Fail to delete worker pod3")
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupB), true)
+	// Different groups do not affect each other.
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupA), false)
+	assert.NoError(t, fakeClient.Delete(ctx, &testPods[0]), "Fail to delete worker pod1")
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, groupA), true)
+}
+
+func TestRayClusterExpectationsDeleteAll(t *testing.T) {
+	ctx := context.TODO()
+	// Simulate local Informer with fakeClient.
+	fakeClient := clientFake.NewClientBuilder().WithRuntimeObjects().Build()
+	exp := NewRayClusterScaleExpectation(fakeClient)
+	namespace := "default"
+	rayClusterName := "raycluster-test"
+	group := "test-group"
+	testPods := getTestPod()
+	exp.ExpectScalePod(namespace, rayClusterName, HeadGroup, testPods[0].Name, Create)
+	exp.ExpectScalePod(namespace, rayClusterName, group, testPods[1].Name, Create)
+	exp.ExpectScalePod(namespace, rayClusterName, group, testPods[2].Name, Delete)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, HeadGroup), false)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, group), false)
+	// Delete all expectations
+	exp.Delete(rayClusterName, namespace)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, HeadGroup), true)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, group), true)
+}
+
+func TestRayClusterExpectationsTimeout(t *testing.T) {
+	ctx := context.TODO()
+	// Reduce the timeout duration so that tests don't have to wait for a long time.
+	ExpectationsTimeout = 1 * time.Second
+	// Simulate local Informer with fakeClient.
+	fakeClient := clientFake.NewClientBuilder().WithRuntimeObjects().Build()
+	exp := NewRayClusterScaleExpectation(fakeClient)
+	namespace := "default"
+	rayClusterName := "raycluster-test"
+	testPods := getTestPod()
+
+	exp.ExpectScalePod(namespace, rayClusterName, HeadGroup, testPods[0].Name, Create)
+	assert.Equal(t, exp.IsSatisfied(ctx, namespace, rayClusterName, HeadGroup), false)
+	// Expectations should be released after timeout.
+	<-time.After(ExpectationsTimeout)
```

Comment:
Why not to use `time.Sleep`? By the way, I think we should wait a little bit longer than the timeout. For example, `ExpectationsTimeout + 1 * time.Second`

Reply:
fixed.